### PR TITLE
Crashed Fixed in Quantity Input Widget

### LINKED ISSUE #1667: App crash on  text input for age on Calculated expression screen
**Describe the bug**

**Component**
SDC library,  SDC gallery app

**To Reproduce**
1. Go to 'Behavior -> calculated expression in Catalog app.'
2. Enter text text in Age input box
3. See error : app crash with java.lang.NumberFormatException

**Expected behavior**
App should not crash

**Screenshots**
logs are are pasted below

**Smartphone (please complete the following information):**
 - Device: all devices
 

**Additional context**

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by PallaviGanorkar**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #1667 

**Description**
Clear and concise code change description. 

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: Bug fix

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

**Comment by PallaviGanorkar** [comment]
For more details
[QuantityWidgetValidation.webm](https://user-images.githubusercontent.com/13957417/196423137-e948eea4-da69-43b3-9313-e961b51e623f.webm)

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemEditTextQuantityViewHolderFactory.kt [hunk 1]
```diff
@@ -37,9 +38,12 @@ internal object QuestionnaireItemEditTextQuantityViewHolderFactory :
       ): QuestionnaireResponse.QuestionnaireResponseItemAnswerComponent? {
         // https://build.fhir.org/ig/HL7/sdc/behavior.html#initial
         // read default unit from initial, as ideally quantity must specify a unit
-        return text
-          .takeIf { it.isNotBlank() }
-          ?.let {
+        return text.let {
+          if (text.isEmpty()) {
+            questionnaireItemViewItem.clearAnswer()
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemEditTextQuantityViewHolderFactory.kt:43]: i don't think you need this line? i think the call site of this function (`getValue`) which is in `QuestionnaireItemEditTextViewHolderFactory.kt` already does this if the return value of this function is null.

so you can remove this line

**Review by jingtang10 — APPROVED** [review]

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemEditTextQuantityViewHolderFactory.kt (hunk 1 above)
Comment (omarismail94) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemEditTextQuantityViewHolderFactory.kt]: @PallaviGanorkar I pushed a commit to fix this. Will merge this PR when checks are complete

Comment (PallaviGanorkar) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemEditTextQuantityViewHolderFactory.kt]: @jingtang10 @omarismail94 Thanks for review and fix it

**Pull request merged**